### PR TITLE
fix: report calculations treat liability balances with wrong sign

### LINKED ISSUE #39: Report calculations treat liability balances with the wrong sign
**Severity:** high
**Category:** code-quality

## Location
- [internal/service/report_service.go:21-33](internal/service/report_service.go#L21-L33)
- [internal/service/report_service.go:212-225](internal/service/report_service.go#L212-L225)

## Problem
Report calculations treat liability balances with the wrong sign.

The domain stores liability balances as negative values. Opening balances follow that rule: a positive liability amount creates a negative split on the liability account. However, reports add liability splits directly into `totalLiabilities` and then compute net worth as `assets - liabilities`.

For a real liability balance of `-2000`, this becomes `assets - (-2000)`, which increases net worth instead of reducing it. Balance sheet totals also render liabilities as negative values, which conflicts with the account-list behavior that strips the sign for liability display.

## Suggested fix
Normalize liability balances before reporting:

- In `GetNetWorthAt`, add liabilities as positive obligations, e.g. `totalLiabilities += utils.AbsInt64(split.Amount)` or subtract the liability balance once consistently.
- In `GenerateBalanceSheet`, store liability report rows and `TotalLiabilities` as positive display/report amounts while preserving the underlying ledger sign in storage.
- Add tests using negative liability balances from opening balances and credit-card spending so net worth decreases as liabilities increase.

### PR DESCRIPTION
**Pull request by Hance08**
## Summary

Closes #39

- **`GetNetWorthAt`**: Changed net worth formula from `assets - liabilities` to `assets + liabilities`, since liability splits are stored as negative values (credit-normal accounts). This correctly subtracts outstanding balances without breaking on mixed-sign splits (charges + payments).
- **`GenerateBalanceSheet`**: Normalized liability balances with `AbsInt64` for display rows and totals, so `TotalLiabilities` and liability row amounts render as positive values (consistent with account list). `NetWorth = assets - TotalLiabilities` then works correctly.
- Updated existing tests to use realistic negative liability values and added new test cases for liability-only currencies and partial payments.

## Test plan
- [x] All `TestGetNetWorthAt` subtests pass (8 cases including new partial payment and liability-only)
- [x] All `TestGenerateBalanceSheet` subtests pass (8 cases with negative liability balances)
- [x] Full test suite `go test ./...` passes with no regressions

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: internal/service/report_service.go [hunk 1]
```diff
@@ -234,8 +236,14 @@ func (ts *TransactionService) GenerateBalanceSheet(ctx context.Context, asOf int
 			result.Assets = append(result.Assets, row)
 			result.TotalAssets[currency] += balance
 		case model.AccountTypeLiability:
+			displayBalance := utils.AbsInt64(balance)
```
Comment (Hance08) [inline internal/service/report_service.go]: This normalization breaks liability accounts that have a debit/positive balance, for example an overpaid credit card. `GetNetWorthAt` now preserves the raw liability sign, so `Assets:Bank=7000` and `Liabilities:Card=+1000` yields net worth `8000`. The balance sheet flips that positive liability through `AbsInt64`, records `TotalLiabilities=1000`, and reports net worth as `7000 - 1000 = 6000`. Since transaction creation only requires balanced splits, this state is reachable via an overpayment. The balance-sheet aggregation needs to preserve the economic sign for net-worth math, or handle negative obligations separately from display formatting.

**Comment by Hance08** [comment]
Addressed the review comment on `AbsInt64` — good catch about overpaid liability accounts.

Fixed in fc1fcd6: switched from `AbsInt64(balance)` to `-balance` (negation). This correctly handles both cases:

- Normal liability (`-3000`) → displays as `3000`
- Overpaid CC (`+1000`) → displays as `-1000`, and `NetWorth = 7000 - (-1000) = 8000` ✓

Added a test case for the overpaid scenario. All tests pass.
